FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,52 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url, completion: { [weak self] result in
+			guard self != nil else { return }
+
+			switch result {
+			case let .success((data, httpResponse)):
+				if httpResponse.statusCode == 200,
+				   let root = try? JSONDecoder().decode(Root.self, from: data) {
+					completion(.success(root.items.map { $0.feedImage }))
+				} else {
+					completion(.failure(Error.invalidData))
+				}
```

Comment:
As a suggestion: it's possible to reduce this method complexity by extracting the mapping logic to a static method, improving readability.
```suggestion
				completion(RemoteFeedLoader.map(data, for: httpResponse))
```